packages/transport/src/create/client.ts

```diff
@@ -0,0 +1,110 @@
+import { createRouterTransport, ServiceImpl, MethodImpl, ConnectRouter } from '@connectrpc/connect';
+
+import {
+  MethodKind,
+  MethodInfo,
+  ServiceType,
+  AnyMessage,
+  JsonValue,
+  Any as xAny,
+} from '@bufbuild/protobuf';
+
+import { TransportMessage, TransportStream, makeSequence } from '../channel/control';
+
+import { InitChannelClientData, InitChannelClientDataType, InitChannelClientMessage } from './init';
+
+import { JsonToMessage } from '../stream';
+
+type CreateAnyMethodImpl<S extends ServiceType> = (
+  methodInfo: MethodInfo & { kind: MethodKind },
+) => typeof methodInfo extends S['methods'][keyof S['methods']]
+  ? MethodImpl<typeof methodInfo>
+  : never;
+
+type CreateAnyServiceImpl = <S extends ServiceType>(
+  service: S,
+  createMethod: CreateAnyMethodImpl<S>,
+) => Partial<ServiceImpl<S>>;
+
+const makeAnyServiceImpl: CreateAnyServiceImpl = <S extends ServiceType>(
+  service: S,
+  createMethod: CreateAnyMethodImpl<S>,
+): Partial<ServiceImpl<S>> => {
+  const impl = {} as { [P in keyof S['methods']]: MethodImpl<S['methods'][P]> };
+  let localName: keyof S['methods'];
+  let methodInfo: MethodInfo;
+  for ([localName, methodInfo] of Object.entries(service.methods))
+    impl[localName] = createMethod(methodInfo);
+  return impl;
+};
+
+type Promised<T> = Parameters<ConstructorParameters<typeof Promise<T>>[0]>;
+
+export const createChannelTransport = (
+  s: ServiceType,
+  requestInit = ({ port, service }: InitChannelClientData): void => {
+    window.postMessage(
+      { type: InitChannelClientDataType, port, service } as InitChannelClientMessage,
+      window.origin,
+      [port],
+    );
+  },
```

Comment:
default request init emits on window, like present. but all further communication happens via the port.